qlib/kernel/arch/aarch64/mm/pagetable.rs

```diff
@@ -132,8 +137,16 @@ bitflags! {
         const ZERO            = 0;
         const PRESENT         = 1;
 
-        const MT_DEGICE_NGNRE = 0x1 << 2;
+        const MT_DEVICE_NGNRE = 0x1 << 2;
         const MT_NORMAL       = 0x4 << 2;
+        // A handy collection of flags that specify device memory
+        const DEVICE_FLAGS    = Self::DEVICE.bits|
+                                Self::UXN.bits|
```

Comment:
add the PXN flag and test to confirm ir works,

Reply:
Added. and it works (on my machine)